app/components/WalletBox.tsx

```diff
@@ -0,0 +1,242 @@
+import { Textarea } from "@nextui-org/input";
+import { Button, Card, Link, Select, SelectItem } from "@nextui-org/react";
+import { useState, useCallback, ChangeEvent, useMemo } from "react";
+import { project_id, user_id } from "../../app_id/app_id.json";
+import { generateWallet } from "../utils/queries";
+import { NETWORK_LIST } from "../utils/networks";
+
+export default function WalletBox() {
+  const [address, setAddress] = useState("");
+  const [network, setNetwork] = useState("");
+  const [balance, setBalance] = useState("");
+  const [onramp_link, setOnrampLink] = useState("");
+  const [offramp_link, setOfframpLink] = useState("");
+  const redirectUrl = "http://localhost:3000/wallet/sell";
+
+  const setNetworkOption = useCallback(
+    (event: ChangeEvent<HTMLSelectElement>) => {
+      setNetwork(event.target.value);
+    },
+    []
+  );
+
+  const generateWalletWrapper = useCallback(async () => {
+    const wallet = await generateWallet({ network_id: network });
+    try {
+      if (wallet) {
+        console.log(wallet.balance);
+        setAddress(wallet.wallet_address);
+        setNetwork(wallet.network_id);
+        setBalance(wallet.balance);
+      } else {
+        setAddress("");
+        setNetwork("");
+        setBalance("");
+      }
+    } catch (error) {
+      alert(error);
+      console.error(error);
+    }
+  }, [network]);
+
+  const generateOnrampLink = useCallback(async () => {
+    let shorten_network = network.split("-")[0];
+    setOnrampLink(
+      `https://pay.coinbase.com/buy/select-asset?appId=${project_id}` +
+        `&addresses={"${address}":["${shorten_network}"]}`
+    );
+  }, [address, network]);
+
+  const generateOfframpLink = useCallback(async () => {
+    let shorten_network = network.split("-")[0];
+    setOfframpLink(
+      `https://pay.coinbase.com/v3/offramp/input?appId=${project_id}` +
```

Comment:
@wen-zhao-cb is this really not /v3/sell? Can we change that before release?

Reply:
It's already changed, I'll update the url